File: 프로그래머스/lv2/12909. 올바른 괄호/올바른 괄호.js
function solution(s){
    let i=0;
    let RightCount=0;
    let LeftCount=0;
    const answer = s.split("")
    
    while(i<s.length){
        answer[i] === "("? LeftCount+=1 : RightCount +=1
        if(RightCount>LeftCount){
            return false
        }
        i++
    }
    return RightCount!==LeftCount?false:true
}

File: 프로그래머스/lv2/12909. 올바른 괄호/올바른 괄호.py
def solution(s):
    i=0
    right_count = 0
    left_count=0
    while(i<len(s)):
        if s[i]=="(":
            left_count +=1 
        else:
            right_count += 1
        if right_count > left_count:
            return False
        i+=1
    if left_count == right_count:
        return True
    return False